builtin/plugin/backend.go

```diff
@@ -49,10 +49,21 @@ func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 
 	sys := conf.System
 
-	// NewBackend with isMetadataMode set to true
-	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, true)
+	// NewBackend with isMetadataMode set to false
+	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, false, true)
 	if err != nil {
-		return nil, err
+		// NewBackend with isMetadataMode set to true
+		raw, err = bplugin.NewBackend(ctx, name, pluginType, sys, conf, true, false)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		b.Backend = raw
+		b.config = conf
+		b.loaded = true
+		b.autoMTLSSupported = true
+
+		return &b, nil
```

Comment:
Should we still have called `raw.Setup()` if this was a plugin with AutoMTLS support?

Reply:
Factory() calls Setup() [here](https://github.com/hashicorp/vault/blob/f0f709478cbd0a92fe17bf66839c6663693c478b/builtin/plugin/backend.go#L33) so I don't think it is necessary.

Reply:
We have to call raw.Setup() below so that we can then immediately call the SpecialPaths() and Type() methods.

Setup() will [initialize the backend](https://github.com/hashicorp/vault/blob/1e8004d810259ff383d9462023ee31e8a4013a27/sdk/plugin/grpc_backend_server.go#L53-L58). Without the call to Setup() we will panic after calling SpecialPaths() [here](https://github.com/hashicorp/vault/blob/1e8004d810259ff383d9462023ee31e8a4013a27/sdk/plugin/grpc_backend_server.go#L106) since the backend hasn't been set.